### PR TITLE
Migrate snapshots publication to GHA

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
We are deprecating [publish-snapshot script ](https://github.com/opensearch-project/opensearch-build/blob/main/publish/publish-snapshot.sh). Hence moving the job to GHA which is much more cleaner and efficient. 

### Issues Resolved
closes https://github.com/opensearch-project/opensearch-testcontainers/issues/75

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/publish-maven-snapshots.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Publish snapshots to maven
+
+on:
+  workflow_dispatch:
+  push:
+      branches:
+        - main
+        - 2.x
```
Comment (reta) [inline .github/workflows/publish-maven-snapshots.yml:8]:
```suggestion
        - 1.x
```

We don't have 2.x (yet) for this project

**Comment by reta** [comment]
Thanks @gaiksaya , just one minor issue on the branching, LGTM otherwise, thank you!

File: .github/workflows/publish-maven-snapshots.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.PUBLISH_SNAPSHOTS_ROLE }}
+          aws-region: us-east-1
+
+      - name: Publish snapshots to maven
+        run: |
+          ./gradlew --no-daemon publishPublishMavenPublicationToSnapshotRepoRepository -PORG_GRADLE_PROJECT_snapshotRepoUsername=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text) -PORG_GRADLE_PROJECT_snapshotRepoPassword=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
```
Comment (gaiksaya) [inline .github/workflows/publish-maven-snapshots.yml:34]: Hi @reta I have not tried this but I feel this is much safer than environment variables. Let me know what you think,

Comment (reta) [reply .github/workflows/publish-maven-snapshots.yml:34]: I have not tried this either, do you see the risks that args like `${aws ... )` could leak raw secrets into logs?

Comment (gaiksaya) [reply .github/workflows/publish-maven-snapshots.yml]: Keeping it what works for everyone for now. Can you please review again? Thanks!

**Review by reta — APPROVED** [review]

**Comment by gaiksaya** [comment]
Sorry forgot to add the credentials. Doing it now. ETA 30min.

**Comment by gaiksaya** [comment]
Added and it published successfully https://github.com/opensearch-project/opensearch-testcontainers/actions/runs/4450761529 too

**Pull request merged**